services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/DocumentTableSchema.java

```diff
@@ -109,12 +112,12 @@ public Map<String, AttributeValue> itemToMap(EnhancedDocument item, boolean igno
     }
 
     private List<AttributeConverterProvider> mergeAttributeConverterProviders(EnhancedDocument item) {
-        List<AttributeConverterProvider> providers = new ArrayList<>();
+        Set<AttributeConverterProvider> providers = new LinkedHashSet<>();
         if (item.attributeConverterProviders() != null) {
```

Comment:
Micro performance improvement suggestion, should we skip copying and just return `attributeConverterProviders` if `CollectionUtils.isEmpty(item.attributeConverterProviders())`?